src/server/frontend_wayland/xdg_shell_v6.cpp

```diff
@@ -433,15 +435,35 @@ void mf::XdgSurfaceV6::unset_maximized()
     }
 }
 
+void mir::frontend::XdgSurfaceV6::clear_next_commit_action()
+{
+    next_commit_action = []{};
+}
+
+void mir::frontend::XdgSurfaceV6::set_next_commit_action(std::function<void()> action)
+{
+    next_commit_action = [this, action]
```

Comment:
[non-blocking] perhaps `next_commit_action` could be a vector so multiple lambdas can be added and executed with no danger of overwriting and not running an action.